packages/app/src/firebaseApp.ts

```diff
@@ -285,21 +299,36 @@ export function createFirebaseNamespace(): FirebaseNamespace {
   /**
    * Create a new App instance (name must be unique).
    */
-  function initializeApp(options: FirebaseOptions, name?: string): FirebaseApp {
-    if (name === undefined) {
-      name = DEFAULT_ENTRY_NAME;
-    } else {
-      if (typeof name !== 'string' || name === '') {
-        error('bad-app-name', { name: name + '' });
-      }
+  function initializeApp(
+    options: FirebaseOptions,
+    config?: FirebaseAppConfig
+  ): FirebaseApp;
+  function initializeApp(options: FirebaseOptions, name?: string): FirebaseApp;
+  function initializeApp(options: FirebaseOptions, rawConfig = {}) {
+    if (typeof rawConfig !== 'object' || rawConfig === null) {
+      const name = rawConfig;
+      rawConfig = { name };
+    }
+
+    const config = rawConfig as FirebaseAppConfig;
+
+    if (config.name === undefined) {
```

Comment:
`typeof config.name === 'undefined'`

Reply:
In this case, checking the `typeof config.name` isn't any different than checking if the value is `undefined`. If this were a top-level symbol, I'd agree.